server/sonar-server/src/main/java/org/sonar/server/qualityprofile/DefinedQProfile.java

```diff
@@ -0,0 +1,127 @@
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.qualityprofile;
+
+import com.google.common.collect.ImmutableList;
+import java.security.MessageDigest;
+import java.util.ArrayList;
+import java.util.List;
+import org.sonar.api.profiles.ProfileDefinition;
+
+import static java.lang.String.format;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.apache.commons.codec.binary.Hex.encodeHexString;
+import static org.apache.commons.lang.StringUtils.lowerCase;
+import static org.sonar.db.loadedtemplate.LoadedTemplateDto.QUALITY_PROFILE_TYPE;
+
+/**
+ * Represent a Quality Profile as computed from {@link ProfileDefinition} provided by installed plugins.
+ */
+public final class DefinedQProfile {
+  private final QProfileName qProfileName;
+  private final boolean isDefault;
+  private final String loadedTemplateType;
+  private final List<org.sonar.api.rules.ActiveRule> activeRules;
+
+  private DefinedQProfile(Builder builder, MessageDigest messageDigest) {
+    this.qProfileName = new QProfileName(builder.getLanguage(), builder.getName());
+    this.isDefault = builder.declaredDefault || builder.computedDefault;
+    this.loadedTemplateType = computeLoadedTemplateType(this.qProfileName, messageDigest);
+    this.activeRules = ImmutableList.copyOf(builder.activeRules);
+  }
+
+  private static String computeLoadedTemplateType(QProfileName qProfileName, MessageDigest messageDigest) {
+    String qpIdentifier = lowerCase(qProfileName.getLanguage()) + ":" + qProfileName.getName();
+    return format("%s.%s", QUALITY_PROFILE_TYPE, encodeHexString(messageDigest.digest(qpIdentifier.getBytes(UTF_8))));
+  }
+
+  public String getName() {
+    return qProfileName.getName();
+  }
+
+  public String getLanguage() {
+    return qProfileName.getLanguage();
+  }
+
+  public QProfileName getQProfileName() {
+    return qProfileName;
+  }
+
+  public boolean isDefault() {
+    return isDefault;
+  }
+
+  public String getLoadedTemplateType() {
+    return loadedTemplateType;
+  }
+
+  public List<org.sonar.api.rules.ActiveRule> getActiveRules() {
+    return activeRules;
+  }
+
+  static final class Builder {
+    private String language;
+    private String name;
+    private boolean declaredDefault;
+    private boolean computedDefault;
+    private List<org.sonar.api.rules.ActiveRule> activeRules = new ArrayList<>();
+
+    public String getLanguage() {
```

Comment:
This getter is used only by the private constructor of DefinedQProfile so it can be dropped.